include/seqan3/alphabet/nucleotide/rna5.hpp

```diff
@@ -56,116 +53,78 @@ namespace seqan3
      * \{
      */
-    //!\copydoc seqan3::dna4::assign_char
-    constexpr rna5 & assign_char(char_type const c) noexcept
-    {
-        using index_t = std::make_unsigned_t<char_type>;
-        _value = char_to_value[static_cast<index_t>(c)];
-        return *this;
-    }
-
-    //!\copydoc seqan3::dna4::assign_rank
-    constexpr rna5 & assign_rank(rank_type const c)
+    constexpr rna5() : base_t{} {}
+    constexpr rna5(rna5 const &) = default;
+    constexpr rna5(rna5 &&) = default;
+    constexpr rna5 & operator=(rna5 const &) = default;
+    constexpr rna5 & operator=(rna5 &&) = default;
+
+    //!\brief Allow implicit construction from dna/rna of the same size.
+    constexpr rna5(dna5 const & r)
     {
-        assert(c < value_size);
-        _value = static_cast<internal_type>(c);
-        return *this;
+        assign_rank(r.to_rank());
     }
     //!\}
 
-    /*!\name Conversion operators
-     * \{
-     */
-    //!\brief Implicit conversion between dna* and rna* of the same size.
-    //!\tparam other_nucl_type The type to convert to; must satisfy seqan3::nucleotide_concept and have the same \link value_size \endlink.
-    template <typename other_nucl_type>
-    //!\cond
-        requires nucleotide_concept<other_nucl_type> && value_size == alphabet_size_v<other_nucl_type>
-    //!\endcond
-    constexpr operator other_nucl_type() const noexcept
-    {
-        return other_nucl_type{_value};
-    }
-
-    //!\brief Explicit conversion to any other nucleotide alphabet (via char representation).
-    //!\tparam other_nucl_type The type to convert to; must satisfy seqan3::nucleotide_concept.
-    template <typename other_nucl_type>
-    //!\cond
-        requires nucleotide_concept<other_nucl_type>
-    //!\endcond
-    explicit constexpr operator other_nucl_type() const noexcept
-    {
-        return detail::convert_through_char_representation<other_nucl_type, std::decay_t<decltype(*this)>>[to_rank()];
-    }
-    //!\}
+protected:
+    //!\privatesection
 
-private:
-    //!\copydoc seqan3::dna4::value_to_char
-    static constexpr char_type value_to_char[value_size]
+    //!\copydoc seqan3::dna4::rank_to_char
+    static constexpr char_type rank_to_char[value_size]
     {
         'A',
         'C',
         'G',
-        'U',
-        'N'
+        'N',
+        'U'
     };
+
+    //!\copydoc seqan3::dna4::char_to_rank
+    static constexpr std::array<rank_type, 256> char_to_rank = dna5::char_to_rank;
+
+    //!\copydoc seqan3::dna4::complement_table
+    static const std::array<rna5, value_size> complement_table;
 };
 
-constexpr rna5 rna5::A{internal_type::A};
-constexpr rna5 rna5::C{internal_type::C};
-constexpr rna5 rna5::G{internal_type::G};
-constexpr rna5 rna5::U{internal_type::U};
-constexpr rna5 rna5::N{internal_type::N};
-constexpr rna5 rna5::T{rna5::U};
-constexpr rna5 rna5::UNKNOWN{rna5::N};
+constexpr std::array<rna5, rna5::value_size> rna5::complement_table
+{
+    rna5::T,    // complement of rna5::A
```

Comment:
shouldn't this be U for consistency/readability? although internally they are the same? If not can't we simply copy the dna5 complement table?

Reply:
I'll have a look!